commons/src/main/java/io/aiven/kafka/connect/common/source/AbstractSourceRecordIterator.java

```diff
@@ -0,0 +1,358 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.source;
+
+import javax.validation.constraints.NotNull;
+
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import org.apache.kafka.connect.data.SchemaAndValue;
+
+import io.aiven.kafka.connect.common.config.SourceCommonConfig;
+import io.aiven.kafka.connect.common.source.input.Transformer;
+import io.aiven.kafka.connect.common.source.input.utils.FilePatternUtils;
+import io.aiven.kafka.connect.common.source.task.Context;
+import io.aiven.kafka.connect.common.source.task.DistributionStrategy;
+import io.aiven.kafka.connect.common.source.task.DistributionType;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import org.apache.commons.io.function.IOSupplier;
+import org.slf4j.Logger;
+
+/**
+ * Iterator that processes cloud storage items and creates Kafka source records. Supports multiple output formats.
+ *
+ * @param <N>
+ *            the native object type.
+ * @param <K>
+ *            the key type for the native object.
+ * @param <O>
+ *            the OffsetManagerEntry for the iterator.
+ * @param <T>
```

Comment:
This should match the letter in AbstractSourceRecord: use `<E>` or `<T>` throughout.